sdk/template/template/test/public/core-v1-test.spec.ts

```diff
@@ -0,0 +1,97 @@
+
+function getRecorderStartOptions(): RecorderStartOptions {
+  // These options are passed to the recorder for playback.
+  const envSetupForPlayback: Record<string, string> = {
+    // In playback, we use the fake SAS URL instead of a real one which would have
+    // been provided via the .env file or otherwise.
+    STORAGE_SAS_URL: fakeSASUrl
+  };
+
+  let sanitizerOptions: SanitizerOptions | undefined;
+
+  if (env.STORAGE_SAS_URL) {
+    const realSASParts = env.STORAGE_SAS_URL.split("/");
+    const fakeSASParts = fakeSASUrl.split("/");
+
+    sanitizerOptions = {
+      // There are several different sanitzers that can be used. A general regex sanitizer
+      // allows for a simple find and replace based on a given regular expression. This is
+      // used here to strip out the sensitive and dynamic parts of the SAS URL in the recording,
+      // replacing them with their corresponding components from the fake SAS URL defined above.
+      generalRegexSanitizers: [
+        // This one replaces the host portion of the URL, which is dynamic based on the storage account...
+        {
+          regex: realSASParts[2],
+          value: fakeSASParts[2]
+        },
+        // ...and this one replaces the query string which contains sensitive information relating to
+        // the SAS token.
+        {
+          regex: realSASParts[3].split("?")[1],
+          value: fakeSASParts[3].split("?")[1]
+        }
+      ]
+    };
+  }
+
+  return {
+    envSetupForPlayback,
+    sanitizerOptions
+  };
+}
+
+describe("Core V1 tests using unified recorder", () => {
+  let recorder: TestProxyHttpClientCoreV1;
+
+  // Note the use of function() instead of an arrow function. This is important
+  // since we pass `this.currentTest` to the client.
+  beforeEach(async function(this: Context) {
+    // Create an instance of recorder. We pass the current test as required by the client;
+    // this value is used to determine the recording's location.
+    recorder = new TestProxyHttpClientCoreV1(this.currentTest);
+
+    // Configure the recorder and start it up prior to each test...
+    await recorder.start(getRecorderStartOptions());
+  });
+
+  afterEach(async () => {
+    // ...and stop it when each test wraps up.
+    await recorder.stop();
+  });
+
+  it("storage-queue create queue", async function() {
+    // Along with specifying the necessary parameters for the QueueServiceClient, we pass through
+    // the test proxy HTTP client, which mean that the client will use the recorder when making requests.
+
+    const client = new QueueServiceClient(env.STORAGE_SAS_URL ?? "", undefined, {
+      httpClient: recorder
+    });
+
+    // Variables can be generated in record mode, and are stored with the recording to be reused
+    // in playback mode.
+    if (!isPlaybackMode()) {
+      recorder.variables["queue-name"] = `queue-${Math.ceil(Math.random() * 1000 + 1000)}`;
+    }
+
+    await client.createQueue(recorder.variables["queue-name"]);
```

Comment:
Personally I think this is a little strange to have to check whether you are in playback mode or not before setting a variable. If it isn't set, do we respect the JS convention and return `undefined`? If so, could we leverage new syntax here:

```suggestion
    recorder.variables["queue-name"] ??= `queue-${Math.ceil(Math.random() * 1000 + 1000)}`;
    
    await client.createQueue(recorder.variables["queue-name"]);
```

Or maybe even a more explicit interface like `variables["queue-name"].init(<value>)` (this has some extra value in that if computing the initial value has side effects, they will still occur even in playback mode, even if the value doesn't need to be initialized).

Reply:
Could even do something like

```ts
const variableName = recorder.variable('variableName', initialValue);
```

to avoid having to index the object. We could even forgo the 'variableName' argument if we guarantee ordering of the calls to `recorder.variable` (and generate the variable IDs by incrementing a counter) a la React hooks:

```ts
const variableName = recorder.variable(initialValue);
```

But maybe that's a bit too much magic.

Reply:
> `recorder.variables["queue-name"] ??= 'queue-${Math.ceil(Math.random() * 1000 + 1000)}';`

@witemple-msft  Just wondering if this would confuse users.

> `const queueName= recorder.variable('variableName', initialValue);`

@timovv this might be a more intuitive one, `.variable()` method would work as a getter in playback mode and a setter in record/live modes.

> `const queueName = recorder.variable(initialValue);`
`await client.createQueue(queueName);`

@timovv this might again confuse users since this would limit users from accessing the same variable using recorder.variable more than once and forces users to save all the variables into separate constants.

Reply:
>this might again confuse users since this would limit users from accessing the same variable using recorder.variable more than once and forces users to save all the variables into separate constants.

If you have to extract each variable into a constant (which I don't think is a bad thing tbh) then nobody will ever have to access the variable using `recorder.variable` more than once anyway :D. As I said, React does something similar with its [`useState` hook](https://reactjs.org/docs/hooks-state.html) and it's quite a nice API in my opinion (and it's prior art).

Reply:
The useState analogy makes sense. `const variable = recorder.variable(<initial value>)` might be my favorite version of this so far, but it's not clear how the variables would be named and stored, so maybe `const variable = recorder.variable(<name>, <initial value>)` makes the most sense.

To me it is really just about making sure that our tests run the exact same code (with all side effects) in playback/record mode so that debugging them is sane.

Reply:
As discussed offline, we're going with option 3 from [here](https://gist.github.com/timovv/354590b29e6a541d42c07e45a634aa98).